lib/rspec/mocks/proxy.rb

```diff
@@ -1,3 +1,5 @@
+RSpec::Support.require_rspec_support 'mutex'
```

Comment:
So the point of this is to be loaded where the core `Mutex` is not available, I'd prefer we replace this with: 
```
unless defined?(Mutex)
  RSpec::Support.require_rspec_support 'mutex'
  Mutex = RSpec::Support::Mutex
end
```

Reply:
👍 I'm glad to do this. The fall-through to stdlib makes sense to me.

All of the other invocations of `RSpec::Support#require_rspec_support` in this project follow the pattern I initially used, so it may be worth filing a ticket to clean them up too.

Reply:
The `require_rspec_support` is just a convenience helper to load a file from our gem, its not always used to polyfill, mostly its actual code we own, it's just this one specific instance, which is only used to create our `ReEntrantMutex` as far as I'm aware.